docs/index.rst

```diff
@@ -28,8 +28,7 @@ be valuable to a wider audience.
     data-model-schema
     schema-changes
     python-development
```

Comment:
Should this page be migrated under the components-release? Although it's listed as development, it mostly documents the release workflow